### DISCUSSION TITLE
Supla integration suggestion

### DISCUSSION THREAD

**Discussion by adamlatacha** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#691] [Ideas]
Hi there,
Recently I'm trying to setup Your design into my Lay-z-spa tub - Cancun model purchased in 2022. I prepared PCB and finally it worked with my tub (I needed to replace blue LLC to red one). 
It's great project - but I'm facing one challenge. I'd like to have full control of my tub out of my home, preferably from my smartphone and also I see opportunities to have cloud-based automation. I'm using many diy smarthome devices based on Supla (https://www.supla.org/) - this is open source smart home platform available on https://github.com/SUPLA.
It allows to operate smart home devices over cloud, mobile app, etc. It also provide native device support for relays, buttons, sensors, thermostats and many more, including MQTT.
I was thinking of incorporating some code modules into Your design but I'm not programmer so it would be kind of "patchy". So I though that maybe You may be interested in using Supla. It might give a way of stop of using local web server and start to use secure. integrated cloud-based solution supported by 3rd party web interface and mobile app.
Starting point might be incorporating simple thermostat, some relays for power, pump bubbles, jets, maybe buttons but only on frontend. All of internal tub display<->esp8266<->CIO communication may be unchanged.

What would be Your thoughts?

Regards
Adam

**Comment by visualapproach** [comment]
Hi Adam, thanks for your suggestion! I'm thinking it's nice to know about this option for global connectivity. This project has required a lot of bending and squeezing to fit on the ESP8266's small RAM. Secure connections require a lot of memory. I withdrew the web page update feature for that reason. Without knowing, I'm not sure it would be worth the time for me to learn how Supla works just to find out that it won't fit in RAM. It appeared to be somewhat heavyweight for this application. Troubleshooting memory issues is NOT fun. I'm not a pro programmer either. So conclusion is that for now I will not go this route. Subject to change.
I can recommend using a VPN to your home network, or using Home assistant/Homey/prometheus to gain access outside local network. Not sure about the last two, but there are other ppl here that do.

/Visualapproach

**Comment by adamlatacha** [comment]
I understand. Then maybe I'll try to incorporate Supla. I believe finally there shouldn't be memory issues even on ESP8266 - as Supla device client is kind of light, removing OTA feature and all web server components will help also. If not - there's always opportunity to change hardware to ESP32.
